src/hooks/usePokemons.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react';
+import { getPokemons } from '../utils/pokemonApi';
+import { BasePokemon } from 'types';
+
+const POKEMON_QTY = 10;
+
+const usePokemons = () => {
+  const [pokemons, setPokemons] = useState<
+    (BasePokemon & { types: string[] })[]
+  >([]);
+
+  useEffect(() => {
+    getPokemons(POKEMON_QTY, 0).then((data) => {
+      setPokemons(data);
+    });
+  }, []);
+
+  const loadMore = (page: number) => {
+    getPokemons(POKEMON_QTY, page).then((data) => {
+      setPokemons((prevState) => [...prevState, ...data]);
+    });
+  };
+
+  return { pokemons, loadMore };
+};
```

Comment:
Es buena practica manejar los loading desde aca. Que el hook retorne 

```suggestion
import { useEffect, useState } from 'react';
import { getPokemons } from '../utils/pokemonApi';
import { BasePokemon } from 'types';

const POKEMON_QTY = 10;

const usePokemons = () => {
  const [pokemons, setPokemons] = useState<
    (BasePokemon & { types: string[] })[]
  >([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    setLoading(true);
    getPokemons(POKEMON_QTY, 0).then((data) => {
      setPokemons(data);
      setLoading(false);
    });
  }, []);

  const loadMore = (page: number) => {
    setLoading(true);
    getPokemons(POKEMON_QTY, page).then((data) => {
      setPokemons((prevState) => [...prevState, ...data]);
      setLoading(false);
    });
  };

  return { pokemons, loadMore, loading };
};
```